### DISCUSSION TITLE
rollbar showing request body as [ ] for put request

### DISCUSSION THREAD

**Discussion by svikramjeet** [rollbar/rollbar-php#515] [Q&A]
Even error are reported to rollbar but can't get request body for PUT request, working fine with POST though

**Comment by bishopb** [comment]
@svikramjeet, could you provide a reproducible example?

Absent further information, I'm guessing that you are using a framework or manually reading the PUT body once already, then attempting to read it again. If this is the case, note that [PHP allows a `PUT` body to be read from `php://input` only once][1]:

> Note: A stream opened with php://input can only be read once; the stream does not support seek operations. However, depending on the SAPI implementation, it may be possible to open another php://input stream and restart reading. This is only possible if the request body data has been saved. Typically, this is the case for POST requests, but not other request methods, such as PUT or PROPFIND.

[1]:https://stackoverflow.com/a/9684732/2908724
